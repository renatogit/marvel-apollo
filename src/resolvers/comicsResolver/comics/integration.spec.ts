import ComicsResolver from '.';
import DataSourcesComicsAPI from '@/dataSources/comics';
import TestServer from '@/configs/test';
import QUERY_GET_COMICS from '@query/comics/queryComics';
import MOCK_GET_COMICS from '@mock/comics/mockComics';

describe('resolvers/comicsResolver/integration', () => {
	it('fetches comics and generate snapshots', async () => {
		const args = {
			dataSources: DataSourcesComicsAPI,
			resolver: ComicsResolver.Query.comics,
			mock: MOCK_GET_COMICS,
			variables: null,
			entity: 'comics',
			typeQuery: 'comics',
		};

		const {server} = await TestServer(args);

		const {body} = await server.executeOperation({
			query: QUERY_GET_COMICS,
		});

		expect(body).toMatchSnapshot();
	});
});
