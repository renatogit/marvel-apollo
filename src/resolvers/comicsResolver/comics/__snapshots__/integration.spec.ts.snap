// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolvers/comicsResolver/integration fetches comics and generate snapshots 1`] = `
{
  "kind": "single",
  "singleResult": {
    "data": {
      "comics": {
        "attributionHTML": "mock data",
        "attributionText": "mock data",
        "code": 200,
        "copyright": "mock data",
        "data": {
          "count": 20,
          "limit": 20,
          "offset": 0,
          "results": [
            {
              "characters": {
                "available": 113,
                "collectionURI": "12744",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 20,
              },
              "collectedIssues": [
                {
                  "name": "mock data",
                  "resourceURI": "mock data",
                },
              ],
              "collections": [
                {
                  "name": "mock data",
                  "resourceURI": "mock data",
                  "role": "mock data",
                },
              ],
              "creators": {
                "available": 102,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                    "role": "mock data",
                  },
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                    "role": "mock data",
                  },
                ],
                "returned": 20,
              },
              "dates": [
                {
                  "date": "mock data",
                  "type": "mock data",
                },
                {
                  "date": "mock data",
                  "type": "mock data",
                },
              ],
              "description": "mock data",
              "diamondCode": "mock data",
              "digitalId": 0,
              "ean": "mock data",
              "events": {
                "available": 113,
                "collectionURI": "12744",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 20,
              },
              "format": "mock data",
              "id": 82967,
              "images": [
                {
                  "extension": "mock data",
                  "path": "mock data",
                },
              ],
              "isbn": "mock data",
              "issn": "mock data",
              "issueNumber": 0,
              "modified": "mock data",
              "pageCount": 112,
              "prices": [
                {
                  "price": 0,
                  "type": "mock data",
                },
              ],
              "resourceURI": "mock data",
              "series": {
                "name": "mock data",
                "resourceURI": "mock data",
              },
              "stories": {
                "available": 161,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 20,
              },
              "textObjects": [
                {
                  "language": "mock data",
                  "text": "mock data",
                  "type": "mock data",
                },
              ],
              "title": "mock data",
              "upc": "mock data",
              "urls": [
                {
                  "type": "mock data",
                  "url": "mock data",
                },
              ],
              "variantDescription": "mock data",
              "variants": [
                {
                  "name": "mock data",
                  "resourceURI": "mock data",
                },
                {
                  "name": "mock data",
                  "resourceURI": "mock data",
                },
              ],
            },
            {
              "characters": {
                "available": 113,
                "collectionURI": "12744",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 20,
              },
              "collectedIssues": [
                {
                  "name": "mock data",
                  "resourceURI": "mock data",
                },
              ],
              "collections": [
                {
                  "name": "mock data",
                  "resourceURI": "mock data",
                  "role": "mock data",
                },
              ],
              "creators": {
                "available": 102,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                    "role": "mock data",
                  },
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                    "role": "mock data",
                  },
                ],
                "returned": 20,
              },
              "dates": [
                {
                  "date": "mock data",
                  "type": "mock data",
                },
                {
                  "date": "mock data",
                  "type": "mock data",
                },
              ],
              "description": "mock data",
              "diamondCode": "mock data",
              "digitalId": 0,
              "ean": "mock data",
              "events": {
                "available": 89.3,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 89.3,
              },
              "format": "mock data",
              "id": 82965,
              "images": [
                {
                  "extension": "mock data",
                  "path": "mock data",
                },
              ],
              "isbn": "mock data",
              "issn": "mock data",
              "issueNumber": 0,
              "modified": "mock data",
              "pageCount": 152,
              "prices": [
                {
                  "price": 0,
                  "type": "mock data",
                },
              ],
              "resourceURI": "mock data",
              "series": {
                "name": "mock data",
                "resourceURI": "mock data",
              },
              "stories": {
                "available": 161,
                "collectionURI": "mock data",
                "items": [
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                  {
                    "name": "mock data",
                    "resourceURI": "mock data",
                  },
                ],
                "returned": 20,
              },
              "textObjects": [],
              "title": "mock data",
              "upc": "mock data",
              "urls": [
                {
                  "type": "mock data",
                  "url": "mock data",
                },
              ],
              "variantDescription": "mock data",
              "variants": [
                {
                  "name": "mock data",
                  "resourceURI": "mock data",
                },
                {
                  "name": "mock data",
                  "resourceURI": "mock data",
                },
              ],
            },
          ],
          "total": 60599,
        },
        "etag": "mock data",
        "status": "mock data",
      },
    },
    "errors": undefined,
  },
}
`;
